src/Relations/BelongsToManyCustom.php

```diff
@@ -84,7 +84,7 @@ private function getIdsWithAttributes($id, $attributes = [])
                     $ids[$attributesArray] = $attributes;
                 }
             }
-        } elseif (is_int($id)) {
```

Comment:
1.is it better to add 

`...} elseif (is_int($id)  ||  is_string($id)) {...`

?

Reply:
Personally I would say not, as you will want a default option.
If I'm honest my coding style is more return early. So I would have if's without else's and return in the if's.
At the end of the day it is your package however.

Reply:
@lewisf2001uk  Actually if someone by accident passes instance that is not Illuminate\Database\Eloquent\Model
child, he will get "Illegal offset type" error then.

Reply:
Yeah, I was just looking at how Laravel / Eloquent deals with it in the `parseIds` function of `InteractsWithPivotTable` trait, they don't do the check of string / int.
I suppose passing back an empty array is better than receiving an exception.

Reply:
If I understood correctly, you agree with me? can you then push that commit so I can merge PR?
PS. I will leave tests as they are in this PR.